site/installation/README.md

```diff
@@ -27,6 +27,47 @@ Verify the version:
 $ kpt version
 ```
 
+## (Optional) enable shell auto-completion
+
+kpt provides auto-completion support for several of the common shells.
+To see the options for enabling shell auto-completion:
+```shell
+kpt completion -h
+```
+
+### Prerequisites
+`kpt` depends on `bash-completion` in order to support auto-completion for the
+bash shell. If you are using bash as your shell, you will need to install
+`bash-completion` in order to use kpt's auto-completion feature.
+`bash-completion` is provided by many package managers
+(see [here][bash-completion]).
+
+### Enable kpt auto-completion
+The kpt completion script for a shell can be generated with the commands
+`kpt completion bash`, `kpt completion zsh`, etc. Sourcing the completion script
+in your shell enables auto-completion.
+
+#### Enable auto-completion for your current shell
+bash:
+```shell
+source <(kpt completion bash)
+```
+zsh:
+```shell
+source <(kpt completion zsh)
+```
+etc.
+#### Enable kpt completion for all your shell sessions
+bash:
+```shell
+echo 'source <(kpt completion bash)' >> ~/.bashrc
+```
+zsh:
+```shell
+echo 'source <(kpt completion zsh)' >> ~/.zshrc
+```
+etc.
```

Comment:
This is great.

I started wondering if we can help users discover the shell automatically. This posts mentions a few ways to get the current shell https://askubuntu.com/questions/590899/how-do-i-check-which-shell-i-am-using

`$0` seems to be contain the shell. See if we can use this ?

Reply:
Thank you for the feedback 😁 

Just to clarify:
Are you suggesting that we include an example in the docs on how to determine the current shell, or that we change the kpt CLI behavior to try to dynamically discover/infer which shell the user wants updated?

Reply:
I was leaning towards `we change the kpt CLI behavior to try to dynamically discover/infer which shell the user wants updated` as the default behavior. So for ex.

```sh
# running the following command configures the autocompletion currently active shell 
source <(kpt completion)

# if one wants to configure auto-completion for shell other than the currently active one
kpt completion <your-fav-shell>
```

Reply:
To summarize our offline discussion:

As the current behavior is the built-in/default behavior of Cobra, and is also the behavior of other cobra-based command line tools (such as [kubectl](https://kubernetes.io/docs/tasks/tools/install-kubectl-linux/#enable-shell-autocompletion)) - we will leave this as-is for now